src/main/scala/utils/KernelUtils.scala

```diff
@@ -0,0 +1,440 @@
+package utils
+
+import nodes.stats._
+import org.apache.spark.rdd.RDD
+import org.apache.spark.broadcast.Broadcast
+import breeze.linalg._
+import scala.collection.mutable.HashMap
+import scala.reflect.ClassTag
+
+object KernelUtils extends Serializable {
```

Comment:
This file shouldn't be there. Almost all of these util functions should be in MatrixUtils in Keystone (else you can add them there)